### PR TITLE
Feature core variational distance update

### PR DESCRIPTION
**Pull request by swenczowski**
In the following, a smaller update / extension of the `VariationalDistanceCalculationProcess` and its MPI equivalent is suggested:

- Motivation: An application made it necessary to use two different processes of type `VariationalDistanceCalculationProcess` at the same time. Inside the C++ implementation of the process, an auxiliary model part called `RedistanceCalculationPart` is created. Also, instructions are present, that re-create or delete this model part. 
Due to the hard-coded naming of model part `RedistanceCalculationPart`, the processes will not distinguish between their own and a different auxiliary model part.

- Problem: Due to this interference between parallely operating distance processes, the one process crashes once the other one deletes or modifies its data.

- Suggested solution: The constructors are equipped with a further optional argument of type "string". Here, the name of an auxiliary model part can be specified. This clear distinction allows to establish several variational distance computations at the same time.

@RiccardoRossi, @rubenzorrilla : This is a preparation step towards the usage of the variational distance process for the two fluid inlet process. Also, it may be helpful once a terrain distance field is introduced.

### CODE REVIEW

File: applications/trilinos_application/custom_processes/trilinos_variational_distance_calculation_process.h [hunk 1]
```diff
@@ -86,9 +86,11 @@ class TrilinosVariationalDistanceCalculationProcess
         ModelPart &base_model_part,
         LinearSolverPointerType plinear_solver,
         unsigned int max_iterations = 10,
-        Flags Options = NOT_CALCULATE_EXACT_DISTANCES_TO_PLANE)
-        : VariationalDistanceCalculationProcess<TDim, TSparseSpace, TDenseSpace, TLinearSolver>(base_model_part, max_iterations, Options),
-        mrComm(rComm)
+        Flags Options = NOT_CALCULATE_EXACT_DISTANCES_TO_PLANE,
+        std::string aux_part_name = "StandardAuxPartName" )
```
Comment (RiccardoRossi) [inline applications/trilinos_application/custom_processes/trilinos_variational_distance_calculation_process.h]: I would keep the same name as before

File: applications/trilinos_application/custom_processes/trilinos_variational_distance_calculation_process.h [hunk 2]
```diff
@@ -86,9 +86,11 @@ class TrilinosVariationalDistanceCalculationProcess
...
+        : VariationalDistanceCalculationProcess<TDim, TSparseSpace, TDenseSpace, TLinearSolver>(base_model_part, max_iterations, Options, aux_part_name ),
+        mrComm(rComm),
+        mAuxModelPart( aux_part_name )
```
Comment (RiccardoRossi) [inline applications/trilinos_application/custom_processes/trilinos_variational_distance_calculation_process.h]: this should be called $mAuxModelPartName$ to tell that it is a string

File: applications/trilinos_application/custom_processes/trilinos_variational_distance_calculation_process.h (hunk 1 above)
Comment (swenczowski) [reply applications/trilinos_application/custom_processes/trilinos_variational_distance_calculation_process.h]: Thank you. Done.

File: applications/trilinos_application/custom_processes/trilinos_variational_distance_calculation_process.h (hunk 2 above)
Comment (swenczowski) [reply applications/trilinos_application/custom_processes/trilinos_variational_distance_calculation_process.h]: Ok, done.

**Comment by swenczowski** [comment]
I am sorry for asking, but are there any other aspects that should be improved? If this is not the case, it would be great if the suggested version could be merged. It will be used both in the InletProcess and in the geometry remeshing with MMG, where also several distance fields are necessary. Thus, having it in the master branch would be a great advantage.

File: kratos/processes/variational_distance_calculation_process.h [hunk 3]
```diff
@@ -123,9 +123,11 @@ class VariationalDistanceCalculationProcess : public Process
         ModelPart& base_model_part,
         typename TLinearSolver::Pointer plinear_solver,
         unsigned int max_iterations = 10,
-        Flags Options = NOT_CALCULATE_EXACT_DISTANCES_TO_PLANE)
-        :mr_base_model_part(base_model_part),
-         mOptions(Options)
+        Flags Options = NOT_CALCULATE_EXACT_DISTANCES_TO_PLANE,
+        std::string aux_part_name = "RedistanceCalculationPart" )
```
Comment (rubenzorrilla) [inline kratos/processes/variational_distance_calculation_process.h]:
```suggestion
        std::string AuxPartName = "RedistanceCalculationPart" )
```
To be consistent with the style guide.

**Review by rubenzorrilla — DISMISSED** [review]
If the tests run both in serial and Trilinos, I've no objections. The behaviour must be the same but now the model part used for the redistance can have a custom name. Am I wrong?

**Comment by swenczowski** [comment]
Yes, you are right. Exactly this change allows several distance processes at one time.

The change considering the naming will be done and I will rerun all unit tests before committing.

**Review by rubenzorrilla — APPROVED** [review]

**Comment by swenczowski** [comment]
bildschirmfoto vom 2019-01-29 15-41-10

For documentation

**Pull request merged**